### PR TITLE
add ut for nodepool controller

### PR DESCRIPTION
**Pull request by charleszheng44**
### Ⅰ. Describe what this PR does
1. remove redundant function
2. Add unit tests for node controller. 

### Ⅱ. Does this pull request fix one issue?
NONE

### Ⅲ. List the added test cases (unit test/integration test) if any, please explain if no tests are needed.
unit tests:
TestConciliateAnnotations
TestConciliateLables
TestConciliateTaints
TestContainTaint
TestIsNodeReady
TestMergeMap
TestRemoveTaint

### Ⅳ. Describe how to verify it
`go test -v github.com/alibaba/openyurt/pkg/yurtappmanager/controller/nodepool`

### Ⅴ. Special notes for reviews
None

### CODE REVIEW

File: pkg/yurtappmanager/controller/nodepool/nodepool_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,552 @@
+/*
+Copyright 2020 The OpenYurt Authors.
```
Comment (Fei-Guo) [inline pkg/yurtappmanager/controller/nodepool/nodepool_controller_test.go:2]: 2020->2021

File: pkg/yurtappmanager/controller/nodepool/nodepool_controller_test.go [hunk 2]
```diff
@@ -0,0 +1,552 @@
...
+					},
+				},
+			},
+			map[string]string{
+				"foo": "bar",
+				"buz": "qux",
+			},
+			map[string]string{"foo": "bar"},
+			map[string]string{"foo": "bar"},
+		},
```
Comment (kadisi) [inline pkg/yurtappmanager/controller/nodepool/nodepool_controller_test.go:57]: 建议再增加一个ut: 主机上有其他的额外label.

		{
			"remove lable",
			&corev1.Node{
				ObjectMeta: metav1.ObjectMeta{
					Name: "test-node",
					Labels: map[string]string{

                                                  "zzzz": "zzzz",
						"foo": "bar",
						"buz": "qux",
					},
				},
			},
			map[string]string{
				"foo": "bar",
				"buz": "qux",
			},
			map[string]string{"foo": "bar"},
			map[string]string{
                             "zzzz": "zzzz",
                           "foo": "bar"
                          },
		},



同样包括： TestConciliateTaints， TestConciliateAnnotations  方法

**Review by kadisi — COMMENTED** [review]
just some nit

**Review by Fei-Guo — COMMENTED** [review]
Nits.

If a file a touched, please update the copyright year.  E.g., nodepool_controller.go

File: pkg/yurtappmanager/controller/nodepool/nodepool_controller_test.go (hunk 1 above)
Comment (charleszheng44) [reply pkg/yurtappmanager/controller/nodepool/nodepool_controller_test.go]: fixed

File: pkg/yurtappmanager/controller/nodepool/nodepool_controller_test.go (hunk 2 above)
Comment (charleszheng44) [reply pkg/yurtappmanager/controller/nodepool/nodepool_controller_test.go:58]: cases added

**Pull request merged**